horovod/run/common/util/safe_shell_exec.py

```diff
@@ -42,24 +44,27 @@ def terminate_executor_shell_and_children(pid):
             pass
 
     # Wait for graceful termination.
-    time.sleep(GRACEFUL_TERMINATION_TIME_S)
+    gone, alive = psutil.wait_procs(p.children(), timeout=GRACEFUL_TERMINATION_TIME_S)
 
     # Send STOP to executor shell to stop progress.
     p.send_signal(signal.SIGSTOP)
 
     # Kill children recursively.
```

Comment:
comment still says `recursively`, can / should we still kill those alive children recursively?

Reply:
Good point.  Added additional logic to get children of the alive processes.